pkg/oci/registry/client.go

```diff
@@ -164,18 +165,27 @@ func BuildImageIDWithTagAndDigest(taggedRef name.Tag, digest digest.Digest) stri
 	return fmt.Sprintf("%s%s%s", taggedRef.String(), DigestDelimiter, digest.String())
 }
 
-func addProxy(transport *http.Transport, proxy string) (*http.Transport, error) {
+func addProxy(transport *http.Transport, proxy string, noProxy string) (*http.Transport, error) {
 	proxyUrl, err := url.Parse(proxy)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	transport.Proxy = func(req *http.Request) (*url.URL, error) {
-		return proxyUrl, nil
+	proxyConfig := httpproxy.Config{
+		HTTPProxy:  proxyUrl.String(),
+		HTTPSProxy: proxyUrl.String(),
```

Comment:
does this work correctly if we always set both http and https proxy independant of yhe actual protocol used?

Reply:
I think it's only about of incoming request, and it should work
https://cs.opensource.google/go/x/net/+/refs/tags/v0.18.0:http/httpproxy/proxy.go;l=127-135
```
func (cfg *config) proxyForURL(reqURL *url.URL) (*url.URL, error) {
	var proxy *url.URL
	if reqURL.Scheme == "https" {
		proxy = cfg.httpsProxy
	} else if reqURL.Scheme == "http" {
		proxy = cfg.httpProxy
		if proxy != nil && cfg.CGI {
			return nil, errors.New("refusing to use HTTP_PROXY value in CGI environment; see golang.org/s/cgihttpproxy")
		}
	}
	if proxy == nil {
		return nil, nil
	}
	if !cfg.useProxy(canonicalAddr(reqURL)) {
		return nil, nil
	}

	return proxy, nil
}

```

no matter if your proxy string has http/or https.

Reply:
yes, the proxyFunc() uses https incase the proxy passed to it has https:// prefix and uses http if it has http:// prefix